### PR TITLE
Use pluginIds instead of currency codes in settings maps

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201840536210680

### CODE REVIEW

File: src/constants/WalletAndCurrencyConstants.js [hunk 1]
```diff
@@ -29,41 +29,42 @@ export const FEE_STRINGS = {
...
-  UFO: s.strings.string_first_ufo_wallet_name,
-  VTC: s.strings.string_first_vertcoin_wallet_name,
-  WAX: s.strings.string_first_wax_wallet_name,
-  XLM: s.strings.string_first_stellar_wallet_name,
-  XMR: s.strings.string_first_monero_wallet_name,
-  XRP: s.strings.string_first_ripple_wallet_name,
-  XTZ: s.strings.string_first_tezos_wallet_name,
-  CELO: s.strings.string_first_celo_wallet_name,
-  SOL: s.strings.string_first_solana_wallet_name
+  'wallet:bitcoincash': s.strings.string_first_bitcoincash_wallet_name,
```
Comment (thehobbit85) [inline src/constants/WalletAndCurrencyConstants.js:32]: Add "initWalletName" prop to SpecialCurrencyInfo

File: src/components/scenes/SendScene.js [hunk 2]
```diff
@@ -657,7 +656,6 @@ export const SendScene = connect<StateProps, DispatchProps, OwnProps>(
       transaction,
       transactionMetadata,
       uniqueIdentifier: guiMakeSpendInfo.uniqueIdentifier,
```
Comment (thehobbit85) [inline src/components/scenes/SendScene.js:659]: Remove this

File: src/constants/WalletAndCurrencyConstants.js [hunk 3]
```diff
@@ -183,9 +183,9 @@ type SpecialCurrencyInfo = {|
   stakeMaxApy?: number
 |}
 
-export const getSpecialCurrencyInfo = (currencyCode: string): SpecialCurrencyInfo => {
-  if (SPECIAL_CURRENCY_INFO[currencyCode]) {
-    return SPECIAL_CURRENCY_INFO[currencyCode]
+export const getSpecialCurrencyInfo = (pluginId: string): SpecialCurrencyInfo => {
```
Comment (thehobbit85) [inline src/constants/WalletAndCurrencyConstants.js:186]: "getPluginId" goes here

File: src/components/scenes/StakingChangeScene.js [hunk 4]
```diff
@@ -201,7 +202,7 @@ export const StakingChangeSceneComponent = (props: Props) => {
       return
     }
 
-    const actionName = SPECIAL_CURRENCY_INFO[currencyCode].stakeActions != null ? SPECIAL_CURRENCY_INFO[currencyCode].stakeActions[change] : ''
+    const actionName = SPECIAL_CURRENCY_INFO[pluginId].stakeActions != null ? SPECIAL_CURRENCY_INFO[pluginId].stakeActions[change] : ''
```
Comment (thehobbit85) [inline src/components/scenes/StakingChangeScene.js:205]: const actionName = SPECIAL_CURRENCY_INFO[pluginId]?.stakeActions[change] ?? ''

File: src/actions/ScanActions.js [hunk 5]
```diff
@@ -315,14 +315,14 @@ export const checkAndShowGetCryptoModal = (selectedWalletId?: string, selectedCu
   try {
     const state = getState()
     const currencyCode = selectedCurrencyCode ?? state.ui.wallets.selectedCurrencyCode
-    const wallets = state.ui.wallets.byId
-    const wallet = wallets[selectedWalletId || state.ui.wallets.selectedWalletId]
+    const { currencyWallets } = state.core.account
+    const wallet: EdgeCurrencyWallet = currencyWallets[selectedWalletId || state.ui.wallets.selectedWalletId]
```
Comment (thehobbit85) [inline src/actions/ScanActions.js:319]: Use '??' instead of '||'

File: src/components/scenes/CryptoExchangeScene.js [hunk 6]
```diff
@@ -388,7 +390,8 @@ export const CryptoExchangeScene = connect<StateProps, DispatchProps, {}>(
         fromWalletPrimaryInfo,
         fromExchangeAmount: div(fromNativeAmount, multiplier, DECIMAL_PRECISION),
         fromFiatToCrypto: getExchangeRate(state, exchangeCurrencyCode, fromIsoFiatCurrencyCode),
-        hasMaxSpend: currencyCode != null && getSpecialCurrencyInfo(currencyCode).noMaxSpend !== true
+        pluginId,
+        hasMaxSpend: pluginId != null && getSpecialCurrencyInfo(pluginId).noMaxSpend !== true
```
Comment (thehobbit85) [inline src/components/scenes/CryptoExchangeScene.js:394]: Remove 'pluginId != null &&'

**Review by thehobbit85 — CHANGES_REQUESTED** [review]

File: src/components/scenes/SendScene.js (hunk 2 above)
Comment (peachbits) [reply src/components/scenes/SendScene.js:659]: also lockinputs and metadata

**Pull request merged**